tools/update_centreon_storage_logs.php

```diff
@@ -0,0 +1,387 @@
+        } elseif ($parameter === '--no-keep') {
+            $shouldDeleteOldData = true;
+        } elseif ($parameter === '--keep') {
+            $shouldDeleteOldData = false;
+        }
+    }
+}
+
+/**
+ * Ask question and wait response of type yes/no
+ *
+ * @param string $question Question to ask
+ * @return bool Return TRUE if response is y or Y
+ */
+function askYesOrNoQuestion(string $question): bool
+{
+    printf("%s [Y] ", $question);
+    $handle = fopen("php://stdin", "r");
+    $response = trim(fgets($handle));
+    $response = $response ?: 'y';
+    fclose($handle);
+    return strtolower($response) === 'y';
+}
+
+/**
+ * Ask question. THe echo of keyboard can be disabled
+ *
+ * @param string $question Question to ask
+ * @param bool $hidden Disable echo of keyboard
+ * @return string Return the response
+ */
+function askQuestion(string $question, bool $hidden = false): string
+{
+    if ($hidden) {
+        system("stty -echo");
+    }
+    printf("%s", $question);
+    $handle = fopen("php://stdin", "r");
+    $response = trim(fgets($handle));
+    fclose($handle);
+    if ($hidden) {
+        system("stty echo");
+    }
+    printf("\n");
+    return $response;
+}
+
+/**
+ * Show message in the standard output
+ *
+ * @param string $message Message to show
+ * @param bool $showStep Set to true if you want showing steps
+ */
+$logs = function (string $message, bool $showStep = true) use (&$currentStep, &$partitionName): void {
+    if ($showStep && $currentStep) {
+        if (! empty($partitionName)) {
+            printf(
+                "[%s] [%s - STEP %d] %s\n",
+                date('Y-m-d H:i:s'),
+                $partitionName,
+                $currentStep,
+                $message
+            );
+        } else {
+            printf(
+                "[%s] [STEP %d] %s\n",
+                date('Y-m-d H:i:s'),
+                $currentStep,
+                $message
+            );
+        }
+    } else {
+        printf("[%s] %s\n", date('Y-m-d H:i:s'), $message);
+    }
+};
+
+/**
+ * Show message but without showing the partition name and step
+ *
+ * @param string $message Message to show
+ */
+function mySysLog(string $message): void
+{
+    global $logs;
+    $logs($message, false);
+}
+
+/**
+ * Stop the Centreon Broker process
+ *
+ * @return int Return status of the Broker stop
+ */
+function stopBroker(): int
+{
+    exec('systemctl stop cbd', $output, $status);
+    return empty($output) ? $status : -1;
+}
+
+/**
+ * Start the Centreon Broker process
+ *
+ * @return int Return status of the Broker start
+ */
+function startBroker(): int
+{
+    exec('systemctl start cbd', $output, $status);
+
+    return empty($output) ? $status : -1;
+}
+
+/**
+ * Retrieve the started process number of Broker
+ *
+ * @return int Return the started process number
+ */
+function getBrokerProcess()
+{
+    exec(
+        'ps -o args -p $(pidof -o $$ -o $PPID -o %PPID -x cbd || echo 1000000) | grep -c /usr/sbin/cbd',
```

Comment:
You use _systemctl_ to start and stop the _cbd_ service. To be consistent you should use _systemctl status_ to detect if the service is running.